kernel/init.c

```diff
@@ -465,6 +466,12 @@ FUNC_NORETURN void z_cstart(void)
 
 	/* perform basic hardware initialization */
 	z_sys_init_run_level(_SYS_INIT_LEVEL_PRE_KERNEL_1);
+
+#if defined(CONFIG_GDBSTUB) && \
+	defined(CONFIG_GDBSTUB_SERIAL_BACKEND)
```

Comment:
need to think how this would be done when we have multiple backends...

Reply:
Yep, don't know exactly how to proceed right now, different backends will probably have different initialization requirements